src/lib/devp_comms.c

```diff
@@ -505,9 +509,16 @@ static void sleep_block_comms (comms_layer_t * p_comms)
 		{
 			if(NULL != mp_sleep_ctrl[i]) // Sleep control is optional
 			{
-				debug1("block %d", i);
+				#ifdef SERVICE_MODE_TIMEOUT
+				m_rcomm_keepalive_timeout = devp_service_mode_timeout_get();
+				#endif //SERVICE_MODE_TIMEOUT
+				if(!m_rcomm_keepalive_timeout)
```

Comment:
Can you guarantee, that 'm_rcomm_keepalive_timeout' is zero by default ?

Reply:
added initialization in init function.